Update docker.md (#6519)
opensearch-project/documentation-website: _install-and-configure/install-opensearch/docker.md

If this is your first time launching an OpenSearch cluster using Docker Compose, use the following example `docker-compose.yml` file. Save it in the home directory of your host and name it `docker-compose.yml`. This file will create a cluster that contains three containers: two containers running the OpenSearch service and a single container running OpenSearch Dashboards. These containers will communicate over a bridge network called `opensearch-net` and use two volumes, one for each OpenSearch node. Because this file does not explicitly disable the demo security configuration, self-signed TLS certificates are installed and internal users with default names and passwords are created.

### Setting a custom admin password

> Starting with OpenSearch 2.12, a custom admin password is required to set up a demo security configuration. For a docker cluster set up using a `docker-compose.yml` file, do either of the following:

Reviewer: ```suggestion
Starting with OpenSearch 2.12, a custom admin password is required to set up a demo security configuration. For a Docker cluster set up using a `docker-compose.yml` file, do either of the following:
```

If this is your first time launching an OpenSearch cluster using Docker Compose, use the following example `docker-compose.yml` file. Save it in the home directory of your host and name it `docker-compose.yml`. This file will create a cluster that contains three containers: two containers running the OpenSearch service and a single container running OpenSearch Dashboards. These containers will communicate over a bridge network called `opensearch-net` and use two volumes, one for each OpenSearch node. Because this file does not explicitly disable the demo security configuration, self-signed TLS certificates are installed and internal users with default names and passwords are created.

> Starting with OpenSearch 2.12.0, a custom admin password is required to set up a demo security configuration. For a Docker cluster-setup using `docker-compose.yml`, you can do either of the following: 1) export `OPENSEARCH_INITIAL_ADMIN_PASSWORD` with a value in the same terminal session before running `docker-compose up`, or 2) create an `.env` file in the same folder as your `docker-compose.yml` file with the `OPENSEARCH_INITIAL_ADMIN_PASSWORD` and strong password values.

Reviewer: I think this would look cleaner as its own section instead of a note. If you opt for this option, remove the `{: note}` on line 159.

```suggestion
### Setting a custom admin password

Starting with OpenSearch 2.12, a custom admin password is required to set up a demo security configuration. For a docker cluster set up using a `docker-compose.yml` file, do either of the following:

1. Export `OPENSEARCH_INITIAL_ADMIN_PASSWORD` with a value in the same terminal session before running `docker-compose up`.
2. Create an `.env` file in the same folder as your `docker-compose.yml` file with the `OPENSEARCH_INITIAL_ADMIN_PASSWORD` and strong password values.
```

### Setting a custom admin password

      - cluster.initial_cluster_manager_nodes=opensearch-node1,opensearch-node2 # Nodes eligible to serve as cluster manager
      - bootstrap.memory_lock=true # Disable JVM heap memory swapping
      - "OPENSEARCH_JAVA_OPTS=-Xms512m -Xmx512m" # Set min and max JVM heap sizes to at least 50% of system RAM

>       - OPENSEARCH_INITIAL_ADMIN_PASSWORD=${OPENSEARCH_INITIAL_ADMIN_PASSWORD}    # Sets the demo admin user password when using demo configuration, required for OpenSearch 2.12 and higher

Reviewer 2: This looks good to me, but I think we may want to call out whether this is set here, or whether this env variable is present from outside or an .env file. Tagging @DarshitChanpura @peterzhuamazon to leave their comments here as well.

Reviewer 3: we should add a note on line 157 stating the requirement to either define the variable in a `.env` and exporting it in the same terminal session where docker-compose is run from.

Reviewer 3: Also, I have a companion PR to update the sample downloadable `docker-compose.yml` file: https://github.com/opensearch-project/project-website/pull/2526

>       - OPENSEARCH_INITIAL_ADMIN_PASSWORD=${OPENSEARCH_INITIAL_ADMIN_PASSWORD}    # Sets the demo admin user password when using demo configuration, required for OpenSearch 2.12 and higher

Reviewer 4: ```suggestion
      - OPENSEARCH_INITIAL_ADMIN_PASSWORD=${OPENSEARCH_INITIAL_ADMIN_PASSWORD}    # Sets the demo admin user password when using demo configuration, required for OpenSearch 2.12 and later
```
